lib/src/config/config.rs

```diff
@@ -588,6 +585,7 @@ pub struct ClientConfig {
 
     /// Database-specific configuration
     ///
+    #[cfg(feature = "database-storage")]
```

Comment:
Shouldn't we gate `storage` as well?

Reply:
No because storage is needed to specify Volatiles environments, and I think this is fine: you could have a database storage and a volatile storage